src/storage/bg_task/segment_sealing_tasks/set_compact_result_segment_status_sealed_task.cpp

```diff
@@ -0,0 +1,69 @@
+//  Copyright(C) 2023 InfiniFlow, Inc. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+module;
+
+#include <vector>
+module set_compact_result_segment_status_sealed_task;
+
+import stl;
+import bg_task;
+import background_process;
+import txn_store;
+import txn_manager;
+import segment_entry;
+import table_entry;
+import buffer_manager;
+import infinity_exception;
+import logger;
+import wal_entry;
+import third_party;
+import build_fast_rough_filter_task;
+import set_segment_status_sealed_task;
+import compaction_alg;
+import compact_segments_task;
+import catalog_delta_entry;
+
+namespace infinity {
+
+void SetCompactResultSegmentStatusSealedTask::CreateAndSubmitTask(TransactionID compact_task_id,
```

Comment:
Compact segment:
0. start new txn
1. merge to new segment
2. create index
3. flush disk
4. commit